src/components/FormErrorHeader.tsx

```diff
@@ -1,12 +1,16 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-condition */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
 /* eslint-disable jsx-a11y/anchor-is-valid */
 import { Alert, List, ListItem } from 'design-system-react';
+import type { FieldErrors } from 'react-hook-form';
 import { Element, Link } from 'react-scroll';
 
 import { FormFieldsHeaderError as formFieldsHeaderError } from 'pages/ProfileForm/types';
+import getAllProperties from 'utils/getAllProperties';
 
 interface FormErrorHeaderProperties {
   id: string;
-  errors: object;
+  errors: FieldErrors | undefined;
```

Comment:
For optional fields I've been doing the following. 
```suggestion
  errors?: FieldErrors;
```